### PR TITLE
Custom proxy support

### PR DESCRIPTION
**Pull request by mvaivre**

### CODE REVIEW

**Comment by mvaivre** [comment]
@nop33 Not yet asking for a review, but some functional testing with a SOCKS5 proxy. @polto Would you like to try this as well? :)

**Comment by nop33** [comment]
I'll have a look into this tmr

**Comment by nop33** [comment]
@mvaivre any instructions on how to perform the functional tests?

File: src/modals/SettingsModal/NetworkSettingsSection.tsx [hunk 1]
```diff
@@ -169,23 +189,46 @@ const NetworkSettingsSection = () => {
...
+            value={tempNetworkSettings.explorerUrl}
+            onChange={(e) => editNetworkSettings({ explorerUrl: e.target.value })}
+            noMargin
+          />
+          <h2 tabIndex={0} role="label">
+            {t('Custom proxy (SOCKS5)')}
+          </h2>
+          <Input
+            id="proxy-address"
+            label={t`Proxy address`}
```
Comment (nop33) [inline src/modals/SettingsModal/NetworkSettingsSection.tsx:221]:
```suggestion
            label={t('Proxy address')}
```

also for below.

File: src/types/settings.d.ts [hunk 2]
```diff
@@ -43,6 +49,8 @@ export interface Settings {
 
 export type Language = 'en-US' | 'fr-FR' | 'de-DE' | 'vi-VN' | 'pt-PT' | 'ru-RU' | 'bg-BG' | 'es-ES' | 'id-ID' | 'tr-TR'
 
+export type Language = 'en-US' | 'fr-FR' | 'de-DE' | 'vi-VN' | 'pt-PT' | 'ru-RU' | 'bg-BG'
```
Comment (nop33) [inline src/types/settings.d.ts:52]: This seems like unintended.

File: public/preload.js [hunk 3]
```diff
@@ -64,6 +64,7 @@ contextBridge.exposeInMainWorld('electron', {
   app: {
     hide: () => ipcRenderer.invoke('app:hide'),
     show: () => ipcRenderer.invoke('app:show'),
-    getSystemLanguage: async () => ipcRenderer.invoke('app:getSystemLanguage')
+    getSystemLanguage: async () => ipcRenderer.invoke('app:getSystemLanguage'),
+    setProxySettings: (proxySettings) => ipcRenderer.invoke('app:setProxySettings', proxySettings)
```
Comment (nop33) [inline public/preload.js:68]: since `setProxySettings` returns a Promise, shouldn't this function be async?

**Review by nop33 — COMMENTED** [review]
I can't get it to work, following the steps I wrote above... I keep getting `ERR_TIMED_OUT` for 
```
GET https://wallet-v20.testnet.alephium.org/infos/chain-params
```

and being stuck in loading state.

File: src/types/settings.d.ts (hunk 2 above)
Comment (mvaivre) [reply src/types/settings.d.ts:52]: Indeed! Weird, and good catch!

File: public/preload.js (hunk 3 above)
Comment (mvaivre) [reply public/preload.js:68]: Mhh, I'm not sure actually. Shouldn't we use the `async` prefix only when using `await` in the function body?
And we don't want to await here I think.

BTW, the "hide" and "show" functions also return a promise, like all the other `invoke` calls.

For consistency sake, we should probably remove all the `async` prefixes in the file.

Comment (nop33) [reply public/preload.js:68]: I think you're right 👍

File: public/electron.js [hunk 4]
```diff
@@ -290,6 +292,17 @@ app.on('ready', async function () {
     if (preferedLanguages.length > 0) return preferedLanguages[0]
   })
 
+  ipcMain.handle('app:setProxySettings', async (_, proxySettings) => {
+    const { address, port } = proxySettings
+    let proxyRules = !address && !port ? undefined : `socks5://${address}:${port}`
```
Comment (nop33) [inline public/electron.js:297]: why `let` and not `const`?

**Review by nop33 — APPROVED** [review]
Apart from that little comment, I don't have any other comments regarding the code or the functionality!

After resolving the conflicts a functional test might be needed again, but I guess nothing will change in the code-wise

**Comment by nop33** [comment]
I think the npm audit is fixed in a following PR, don't bother in this one.

**Comment by mvaivre** [comment]
Note for myself: one more thing todo is add a little "experimental" badge next to the feature.

**Comment by mvaivre** [comment]
@nop33 asking for a quick re-review 🙏

**Review by nop33 — APPROVED** [review]

**Comment by nop33** [comment]
Functional tests are good ✅  Will merge once tests pass.

**Pull request merged**